src/Microsoft.Extensions.FileProviders.Physical/PhysicalFilesWatcher.cs

```diff
@@ -137,54 +190,47 @@ private void ReportChangeForMatchedEntries(string pattern)
             }
         }
 
-        private string NormalizeFilter(string filter)
+        private static string NormalizePath(string filter) => filter.Replace(Path.DirectorySeparatorChar, '/');
+
+        private static bool IsDirectoryPath(string path)
         {
-            // If the searchPath ends with \ or /, we treat searchPath as a directory,
-            // and will include everything under it, recursively.
-            if (IsDirectoryPath(filter))
+            return path.Length > 0
```

Comment:
I see that the old code was doing a null check too. Did you remove it by mistake?